matrix/lambdas/daemons/driver.py

```diff
@@ -43,21 +43,30 @@ def run(self, bundle_fqids: typing.List[str], bundle_fqids_url: str, format: str
             resolved_bundle_fqids = bundle_fqids
         logger.debug(f"resolved bundles: {resolved_bundle_fqids}")
 
+        request_hash = RequestCache(self.request_id).set_hash(resolved_bundle_fqids, format)
+        logger.debug(f"Calculated and set hash {request_hash} for request {self.request_id}")
+        request_tracker = RequestTracker(request_hash)
+
+        if request_tracker.is_initialized:
+            logger.debug(f"Halting because {request_hash} already exists.")
+            return
```

Comment:
So if the request hash exists, but it failed, we still halt and don't retry the request. Do we want to retry failed requests?

Reply:
I'm not sure. I guess we don't know how many failed requests can be retried. I'd _like_ to think it's not too many.

Reply:
Okay I added a little change that I think should retry. Let me know if that's what you were thinking.

Reply:
Yup that addressed it